app/api.py

```diff
@@ -345,3 +348,78 @@ async def tech_stack_graph():
     return json.loads(tech_stack_by_role(df).to_json())
+
+
+@API.post("/query/Feedback")
+async def feedback_crud(crud: str, mentee_id: Optional[str] = None, mentor_id: Optional[str] = None,
+                        feedback: Optional[str] = None, ticket_id: Optional[str] = None):
+    """Create, read, update and delete records in the feedback collection.
+
+    Creates new document within Feedback using the data parameter to
+    populate its fields.
+
+    Args:
+    - crud (create, read and delete): choice (pick 1)
+    - ticket_id (uuid) : autogenerated 16 digit ID (optional)
+    - mentee_id (str): Mentee ID (optional)
+    - mentor_id (str): Mentor ID (optional)
+    - feedback (str): feedback (optional)
+    - date_time (datetime): current datetime
+    - vader_score(str): compound sentiment value
+
+    Returns:
+    - if crud == create -> supply mentee_id, mentor_id and feedback
+    - if crud == read -> supply ticket_id or mentor_id to view specific ticket or None to view all tickets
+    - if crud == delete -> supply ticket_id to delete ticket
+    """
+
+    if crud == 'create' and (mentee_id, mentor_id, feedback):
+        feedback_score = vader_score(feedback)
+        return {"result": API.db.create("Feedback", {"ticket_id": generate_uuid(16),
+                                                     'mentee_id': mentee_id, 'mentor_id': mentor_id,
+                                                     'feedback': feedback, 'datetime': datetime.datetime.now(),
+                                                     'vader_score': feedback_score})}
+
+    elif crud == 'read':
+        if ticket_id:
+            return {"result": API.db.read("Feedback", {'ticket_id': ticket_id})}
+        elif mentor_id:
+            return {"result": API.db.read("Feedback", {'mentor_id': mentor_id})}
+        else:
+            return {"result": API.db.read("Feedback", ticket_id)}
+
+    elif crud == 'delete':
+        API.db.delete("Feedback", {"ticket_id": ticket_id})
+        return {"result": {"deleted": ticket_id}}
+
+    else:
+        return 'Modify_option needs to be create, read or delete'
+
+
+@API.get("/graphs/tech-stack-by-role")
+async def tech_stack_graph():
+
+    df = pd.DataFrame(feedback_crud('read'))
+    return json.loads(feedback(df).to_json())
+
+
+@API.get("/graphs/feedback_window")
+async def mentor_feedback():
+    """create the dataframe for visualization"""
+    df = pd.DataFrame(API.db.read('Feedback'))
+    df['datetime'] = np.random.choice(
+                        pd.date_range('2020-01-01', '2022-01-01'),
+                        len(df))
+    df['vader_score'] = df['feedback'].apply(vader_score)
+    return json.loads(feedback_window(df).to_json())
+
+
+@API.get("/graphs/mentor_feedback_individual")
+async def mentor_feedback_progress():
+    """create the dataframe for visualization"""
+    df = pd.DataFrame(API.db.read('Feedback'))
+    df['datetime'] = np.random.choice(
```

Comment:
A random datetime here is problematic. We should update the Feedback generator class with this new field instead.